Update documentation with audit comments (#20)
calimero-is-near/docs: docs/bridge/bridging/1_prerequisites.mdx

---
title: Prerequisite for using the bridge
sidebar_position: 1
---

1. A Calimero shard needs to be running (in case you want to spin up your own Calimero shard send an [email](mailto:info@calimero.network) or [fill out this form](https://www.calimero.network/contact))
2. All contracts need to be deployed on both NEAR and Calimero
3. Relayer from Near to Calimero and a relayer from Calimero to Near need to be running
4. Bridge service that monitors events in realtime on both Near and Calimero need to be running

### Prerequisites for bridging FTs/NFTs

> When the Account is bridging a token (FT or NFT) from the source to the destination chain (for example from Calimero to Near) it should always obtain

Reviewer: ```suggestion
When an Account is bridging a token (FT or NFT) from the source to the destination chain (for example from Calimero to Near) it should always obtain
```

> the same Account ID on the destination chain. Let us show how Bob can lose his tokens if he does not obtain a destination account before bridging.

Reviewer: ```suggestion
the same Account ID on the destination chain beforehand. Let us show how Bob can lose his tokens if he does not obtain a destination account before bridging.
```

Bob has Account ID bob.testnet on the Calimero side and wants to bridge his FTs to the Near side. Bob successfully locks his FTs on the
FT Connector on the Calimero side. Now Alice observes that Bob is bridging his assets to the Near side and creates bob.testnet Account on
Near side (Bob did not possess this Account ID on the Near side before bridging). Now Alice gains Bob's bridged FTs and has successfully stolen them. Note that here Alice did not have any information that is not publicly available on Near side, so this problem can not be fixed by properly setting permissions.

> If you want to securely bridge assets from the source chain to the destination chain you should first possess the same Account ID on both sides, and

Reviewer: ```suggestion
If you want to securely bridge assets from the source chain to the destination chain you should *be in possession of the same Account ID on both sides*, and
```
See if this will bold the text to emphasize importance

Author: Fixed, `**` is for bolding the text, that is the reason I did not just commit this suggestion

if that is not possible for some Account ID you should find another Account ID and obtain that new Account ID on both sides. Now you should

> bridge assets from a newly obtained Account ID.

Reviewer: ```suggestion
```

:::caution

> When bridging tokens you **MUST** possess the Account ID on the destination side. Assets **CAN** easily be stolen if you try to obtain the destination side Account ID after the bridging.

Reviewer: ```suggestion
When bridging tokens you **MUST** possess the Account ID on the destination side. Assets **CAN** be easily lost if you try to obtain the destination side Account ID after the bridging.
```
